python/src/read.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef NJOY_NDITK_PYTHON_READ
+#define NJOY_NDITK_PYTHON_READ
+
+// system includes
+#include <string>
+
+// other includes
+#include "tools/Log.hpp"
+#include "tools/disco/FreeFormatCharacter.hpp"
+
+/**
+ *  @brief Read a record from a string
+ *
+ *  @param[in] string   the string to read from
+ */
+template < typename Record, typename... Arguments >
+Record read( const std::string& string, Arguments... arguments ) {
+
+  using namespace njoy::tools;
+
+  Record record;
+  auto iter = string.begin();
+  auto end = string.end();
+
+  std::string key = disco::FreeFormatCharacter::read< std::string >( iter, end );
+  if ( record.keyword() == key ) {
+
+    //! @todo take into account _0 style keys
+    record.read( iter, end, arguments... );
+
+    //! @todo verify the string is now empty
```

Comment:
So read leaves the std::string string empty?

Reply:
Well, it should.